dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java

```diff
@@ -1174,21 +1175,39 @@ private JSONArray getAvailableWorkflowActionsListingJson(final User currentUser,
 
 	@CloseDB
 	public ArrayList<String[]> doSearchGlossaryTerm(String valueToComplete, String language) throws Exception {
-		ArrayList<String[]> list = new ArrayList<String[]>(15);
-
-		List<LanguageKey> props = retrieveProperties(Long.parseLong(language));
-
+		final int limit = Config.getIntProperty("glossary.term.max.limit",15);
+		ArrayList<String[]> list = new ArrayList<String[]>(limit);
+		final User systemUser = APILocator.systemUser();
+		final long languageId = Long.parseLong(language);

```

Comment:
you may use ConversionUtils.toLong (language) it is safer, well it depends if you are expecting the exception

Reply:
Not expecting an error here since it gets the languageId directly from the content, so not possible to get a wrong languageId.